### ISSUE TITLE
Give the engineer a better secondary grenade

### ISSUE THREAD

**Issue by AfterShockFF** [fortressforever/fortressforever#164]
Preferably one that doesnt teamkill everyone around him, detonate nearby pipes and instagib all his buildables.

Maybe something that could cover him somewhat whilst he builds, or something that produces metal, repairs his buildings for him, etc.

**Comment by WiFiDi** [comment]
an idea i had awhile back is to instead of removing all the grenades to make each class hvae 2 secondary grenades to choose from. (in the class selection menu) shame to to completely get rid of all the nades in the game.

for instance a heavy has sowfield they had to get rid of demo nade (forgot what its is called demos current secondary nade like it used to have.) why not give players the option to pick which grenade it wants. ff already has alot of divide between offense and defense roles why try to design a nade for both instead of just making 2 seems easier and makes each class marginally more flexible. youd have to make some new grenades but that way each class could have a offensive and defensive grenade. that way a heavy playing offense doesn't have to use a slowfield (in pubs) and solider maybe can have something better for offense like the nail grenade or maybe something else.

**Comment by WiFiDi** [comment]
id also like to note id rather not start removing the grenades we already have we don't have that many left and new ones havn't created universly better soloution. id rather look towards a different soloution so it didn't come down remvoing weapons and nades to get that people are used to.and have fun with.

**Comment by AirTrek** [comment]
Don't take away the EMPwn ; ;. It's dangerous and deadly (not as deadly as it used to be) and therefore it makes sense that it should be used with caution. I think the spy deserves to get a replacement grenade.

**Comment by fpsmoto** [comment]
I, too, am against replacing or removing the EMP. It's already a great grenade.

**Comment by AfterShockFF** [comment]
It might be great in AvD or public servers but it's actually quite self-destructive in competitive / CTF games and can harm the team a fair amount when used. Many new players playing engineer accidentally teamkill their team, demoman and destroy his pipetraps and will have a very bad time when trying to use the EMP. It also does next to nothing against medics and scouts which comprise the majority of the offense, and if dropped near the SG (where the engineer is the majority of the time) will kill his SG and dispenser and destroy any nearby metal bags, preventing him from rebuilding quickly upon return.

Additionally, the engineer has problems recovering in the flag room when it is overrun, and grenades are a great way to provide a self-balancing mechanic to cover this without overpowering the class, enabling each player to get initial power which weakens over time until he dies. This gap could be reduced by a decent grenade. 

I'm sure we can find a use for the EMP outside of the engineer class, ideally on a class where killing buildables / pipetraps is more "in role". In particular, its currently the only class in the game capable of destroying a pipetrap - something offense teams should be capable of doing without using an offense engineer.

**Comment by fpsmoto** [comment]
Many new players also play in pubs, where friendly fire is usually off. Sure they can accidentally kill their teammate's pipes, but that's all in the learning process. You can't expect someone new to know what they're doing and tbh, even in pickups the use of EMPs isn't a bad thing. The primary reason it doesn't do much damage to scouts or medics is because they've already likely thrown their backpacks down to lower the # of metal they have, but this can be done on defense too to limit the damage taken from a friendly EMP. I think that lack of easy-to-understand guides or tutorials would be more cause for its misuse than anything else. We have an FF training mode, but we need to build on it and be more specific to each class so new players can be taken through numerous scenarios so they have a better understanding of how grenades like this work. I mean MIRVs, among other grenades, can do a helluva lot of damage to a friendly team as well. Does that mean they need to be replaced? I don't buy that as a valid reason.

Could the grenade be made better? Perhaps, such as using it to temporarily disable a buildable or cause it to malfunction or shoot in random directions, but it's a pretty powerful grenade as is and I doubt more than a few people would be happy with it removed, even including a lot of pickup players.

TBH, the best thing to focus on before try to change existing grenades that are actually a decent weapon in the game, is to work on classes that don't even have a primary or secondary grenade such as Sniper, Spy and Scout.

**Comment by Xylemon** [comment]
I have to agree that the grenade seems fine. I also have felt that FF does also seem a bit lacking without the classic Spy and Scout secondary nades.

**Comment by WillWowmc** [comment]
On that, are new grenades for Spy and Scout a thing being looked into for the future? Also, was the gas grenade removed due to the hallucinations being too confusing for some players? I think that a more dynamic hint system, which alerts newer players to status effects and such would eliminate some confusion on that matter. (By the way, I'm new to github, so please excuse me if I'm 'necroposting', saying something possibly off-topic or something else in git-hub culture)

**Comment by squeek502** [comment]
We've never really had an idea for a replacement scout grenade. I'm actually not totally opposed to bringing caltrops back, but the old FF implementation definitely wasn't good (canisters that were primed like normal grens rather than instant deployment like in TFC).

The gas grenade was removed because they are a pure spam grenade with little to no interactivity/depth. Something I found on the forums that I had said about them:

> the problems with gas grenades: there is no risk to throwing them; it doesn't matter where you throw them really; the enemy is almost always forced to run into them; there is no counter except waiting and/or going another route (which obviously isn't possible if they are spamming a spawn).
